compiler/optimizer/RegisterCandidate.cpp

```diff
@@ -2249,7 +2249,7 @@ static void assign_candidate_loop_trace_increment(TR::Compilation *comp, TR_Regi
 bool
 TR_RegisterCandidates::assign(TR::Block ** cfgBlocks, int32_t numberOfBlocks, int32_t & lowestNumber, int32_t & highestNumber)
    {
-#if (defined(__IBMCPP__) || defined(__IBMC__))
+#if (defined(__IBMCPP__) || defined(__IBMC__)) && (!defined(LINUXPPC))
```

Comment:
I think you want to check for `!defined(__ibmxl__)` here? (Since whoever put this in was suggesting that it was a workaround for old XLC on Linux and it to continue to work in that case.)